infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/workflows/RemoveNodeWorkflow.java

```diff
@@ -0,0 +1,75 @@
+package org.corfudb.infrastructure.orchestrator.workflows;
+
+import com.google.common.collect.ImmutableList;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.orchestrator.Action;
+import org.corfudb.infrastructure.orchestrator.ActionStatus;
+import org.corfudb.infrastructure.orchestrator.IWorkflow;
+import org.corfudb.protocols.wireprotocol.orchestrator.RemoveNodeRequest;
+import org.corfudb.protocols.wireprotocol.orchestrator.Request;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.view.Layout;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import static org.corfudb.protocols.wireprotocol.orchestrator.OrchestratorRequestType.REMOVE_NODE;
+
+/**
+ * A workflow definition that removes a node from the cluster, if it exists. This workflow
+ * will fail if the remove results in a cluster that is not redundant.
+ * @author Maithem
+ */
+
+@Slf4j
+public class RemoveNodeWorkflow implements IWorkflow {
+
+    final RemoveNodeRequest request;
```

Comment:
either make it private or document it and add a getter.

Reply:
Done.